src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -730,7 +730,8 @@ FromClauseRecurringTupleType(Query *queryTree)
 		return RECURRING_TUPLES_EMPTY_JOIN_TREE;
 	}
 
-	if (FindNodeCheckInRangeTableList(queryTree->rtable, IsDistributedTableRTE))
+	if (FindNodeCheckInRangeTableList(queryTree->rtable,
+									  IsDistributedNonReferenceTableRTE))
```

Comment:
Do we have a test to validate this change? If not, please add one.

Reply:
This isn't changed. I renamed `IsDistributedTableRTE` to `IsDistributedNonReferenceTableRTE` since that's what it actually was implemented as. grep for `IsDistributedTableRTE` for which cases were truly updated

Reply:
maybe
DistributedNonReferenceTable -> DistributedTable
DistributedTable -> CitusTable

Reply:
Still confusing unless we rename `IsDistributedTable` to `IsCitusTable`

Reply:
That's what I meant